.tekton/open-infra-deployment-pr-osp-nightly.yaml

```diff
@@ -18,8 +18,12 @@ spec:
     - name: infra-deployment-update-script
       value: |
         sed -i -E 's/[0-9a-f]{40}/{{ revision }}/g' components/pipeline-service/development/kustomization.yaml
+        # just pin the watcher image, which is the second image ref, so use 2 vs. g
+        sed -i -E 's/newTag: {{ revision }}/newTag: bae7851ff584423503af324200f52cd28ca99116/2' components/pipeline-service/development/kustomization.yaml
```

Comment:
If I understand correctly what you are trying to achieve, that sed expression won't do the trick. `s/foo/bar/2` replaces the second occurrence on the same line, not the second occurrence in the file.
```
-> cat /tmp/cat 
cat
cat dog
cat dog
cat dog dog
-> sed 's/dog/mouse/2' /tmp/cat 
cat
cat dog
cat dog
cat dog mouse
```
The easiest thing I can think of is using `yq`:
```
-> cat /tmp/k.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - https://github.com/openshift-pipelines/pipeline-service.git/operator/gitops/argocd/pipeline-service?ref=72287aca6503f631b917debc27683a508f7e45ad

images:
  - name: quay.io/redhat-appstudio/tekton-results-api
    newTag: bae7851ff584423503af324200f52cd28ca99116
  - name: quay.io/redhat-appstudio/tekton-results-watcher
    newTag: bae7851ff584423503af324200f52cd28ca99116
[ foo] -> yq e -i '.images[1].newTag = 111' /tmp/k.yaml
[ foo] -> cat /tmp/k.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - https://github.com/openshift-pipelines/pipeline-service.git/operator/gitops/argocd/pipeline-service?ref=72287aca6503f631b917debc27683a508f7e45ad
images:
  - name: quay.io/redhat-appstudio/tekton-results-api
    newTag: bae7851ff584423503af324200f52cd28ca99116
  - name: quay.io/redhat-appstudio/tekton-results-watcher
    newTag: 111
```
If you do not have `yq`, then you can do with `awk`, but the expression is much more complex.

Reply:
thanks so much @enarha  .... I do have both `yq` and `awk` but of course will give your precise example a go and update

Reply:
The yq should exist on the image used for that task. It does in most cases where I needed it.

Reply:
fyi @enarha your 
`yq -e -i '.images[1].newTag = 111' ./operator/gitops/argocd/pipeline-service/tekton-results/kustomization.yaml ` worked as is, but when when I substituted `111` with `bae7851ff584423503af324200f52cd28ca99116` I had to surround `bae7851ff584423503af324200f52cd28ca99116` with double quotes.

Specifically
```
gmontero ~/go/src/github.com/openshift-pipelines/pipeline-service  (repin-watcer)$ yq -e -i '.images[1].newTag = "bae7851ff584423503af324200f52cd28ca99116"' ./operator/gitops/argocd/pipeline-service/tekton-results/kustomization.yaml 
gmontero ~/go/src/github.com/openshift-pipelines/pipeline-service  (repin-watcer)$ 
```

Otherwise I got this error:
```
gmontero ~/go/src/github.com/openshift-pipelines/pipeline-service  (repin-watcer)$ yq e -i '.images[1].newTag = bae7851ff584423503af324200f52cd28ca99116' ./operator/gitops/argocd/pipeline-service/tekton-results/kustomization.yaml 
Error: 1:21: invalid input text "bae7851ff5844235..."
gmontero ~/go/src/github.com/openshift-pipelines/pipeline-service  (repin-watcer)$ 
```

will update the PR shortly ... thanks again for the catch / suggestion / fix